### PR TITLE
Make CLI option

### LINKED ISSUE #9: Ruby: Add an executable
My understanding of Ruby is *very* limited, but I believe a gem can be made executable. I propose doing so for this project so we could run the username generator locally in the CLI.

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Added an executable CLI option - can be called with `do_username`.

Closes #9

### CODE REVIEW

File: bin/do_username [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env do_username
+
+require 'do_username'
+if ARGV.length < 1
+    puts DOUsername.generate()
```
Comment (brunoocasali) [inline bin/do_username]: You don't need the empty parenthesis `()` when the method call doesn't have arguments ;)

```suggestion
  puts DOUsername.generate
```

File: bin/do_username [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    exit
+end
+
+puts DOUsername.generate(ARGV[0].to_i)
```
Comment (brunoocasali) [inline bin/do_username]: You could add a empty line at the end of this file! (is a git good practice - https://thoughtbot.com/blog/no-newline-at-end-of-file) ;)

File: bin/do_username [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env do_username
```
Comment (brunoocasali) [inline bin/do_username]: What about?

```suggestion
#!/usr/bin/env do_username

require 'do_username'

max_size = ARGV[0].to_i

if max_size > 0
  puts DOUsername.generate(max_size)
else
  puts DOUsername.generate
end
```

File: bin/do_username (hunk 2 above)
Comment (naomi-lgbt) [reply bin/do_username]: Oops! Usually I do that lol

File: bin/do_username (hunk 1 above)
Comment (naomi-lgbt) [reply bin/do_username]: Coming from JS, that's new to me 😮

File: bin/do_username [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
```
Comment (MattIPv4) [inline bin/do_username]:
```suggestion
#!/usr/bin/env ruby
```
This should be running as ruby?

File: bin/do_username [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+require 'do_username'
+
+max_size = ARGV[0].to_i
+
+if max_size > 0
+  puts DOUsername.generate(max_size)
+else
+  puts DOUsername.generate
+end
```
Comment (MattIPv4) [inline bin/do_username]:
```suggestion
max_size = begin
             Integer(ARGV[0])
           rescue ArgumentError
             ARGV[0]
           end

puts DOUsername.generate(max_size)
```
I think we should let the library itself handle validation (rebasing this PR against master should help with that), so within this executable we should cast strictly to an int or pass the given argument directly through

**Review by MattIPv4 — CHANGES_REQUESTED** [review]
The bin file isn't executable, `chmod a+x bin/do_username` should fix that. The new file permissions can be comitted.

**Comment by naomi-lgbt** [comment]
Okay, so I pulled down the recommended changes, rebased against `master`, and gave the CLI a test. Calling `do_username` with no argument throws a TypeError and crashes (cannot convert `nil` to `Integer`). I played around with the binary and was able to rescue both the TypeError and Argument error, but had to change the fallback to a hard-coded `30` as using `ARGV[0]` still errored out. Before I commit, is there a different approach you'd prefer?

```ruby
#!/usr/bin/env ruby

require 'do_username'

max_size = begin
             Integer(ARGV[0])
           rescue TypeError
             30
           rescue ArgumentError
             30
           end

puts DOUsername.generate(max_size)
```

Also, permission fix should be committed. Turned out I missed a `git` config setting to include file perms in the commits.

**Comment by MattIPv4** [comment]
@nhcarrigan could you rebase this against master (all the ruby stuff is now in `ruby/`)?

My thinking (to handle nil) is something like:

```
#!/usr/bin/env ruby

require 'do_username'

arg = ARGV[0]

if arg.nil? || arg.empty?
  puts DOUsername.generate
else
  puts DOUsername.generate(Integer(arg))
end
```

This let's `do_username` work, and for the CLI to essentially handle validating the input as an integer, but everything else is handled by the library itself still.

We should not silently handle errors and default to `30`, we should surface errors if invalid inputs are given.

**Comment by naomi-lgbt** [comment]
Alright, gave this another rebase -  applied your suggested approach, executable seems to be working as intended. :)

File: README.md [hunk 6]
```diff
@@ -59,6 +59,21 @@ https://www.npmjs.com/package/do_username
...
+
+# Call the CLI command with a max length of 10 characters
+$ do_username 10 
+```
+
+```text
+GiantBlueGreenScubaDiver
+BlueMorel
+```
+
```
Comment (MattIPv4) [inline README.md]: Can this be added to ruby/README.md please, instead of here?

**Review by MattIPv4 — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (naomi-lgbt) [reply README.md]: Sure thing! Sorry - totally forgot I added instructions to the readme. 😅

**Review by MattIPv4 — APPROVED** [review]
Lgtm, ty!

**Pull request merged**